src/main/java/org/folio/service/rollover/UniqueValidationService.java

```diff
@@ -0,0 +1,53 @@
+package org.folio.service.rollover;
+
+import static org.folio.dao.rollover.LedgerFiscalYearRolloverDAO.LEDGER_FISCAL_YEAR_ROLLOVER_TABLE;
+import static org.folio.rest.persist.HelperUtils.getFullTableName;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dao.rollover.LedgerFiscalYearRolloverDAO;
+import org.folio.rest.jaxrs.model.LedgerFiscalYearRollover;
+import org.folio.rest.persist.DBClient;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.ext.web.handler.HttpException;
+
+public class UniqueValidationService {
+
+  private static final Logger log = LogManager.getLogger(UniqueValidationService.class);
+  private final LedgerFiscalYearRolloverDAO ledgerFiscalYearRolloverDAO;
+
+  public UniqueValidationService(LedgerFiscalYearRolloverDAO ledgerFiscalYearRolloverDAO) {
+    this.ledgerFiscalYearRolloverDAO = ledgerFiscalYearRolloverDAO;
+  }
+
+  public Future<Void> validationOfUniqueness(LedgerFiscalYearRollover rollover, DBClient client) {
```

Comment:
name of this method should be different as previously discussed.

Reply:
Fixed. Metod renamed to checkRolloverExists.